files/default/neutron-ha-tool.py

```diff
@@ -913,6 +922,14 @@ def pick(self):
         return random.choice(self.agents)
 
 
+class Configuration(object):
+    """
+    Registry for storing application's configuration
+    """
+
+    agent_picker_class = RandomAgentPicker
```

Comment:
Hmm.  This isn't really much more than a glorified global variable which just happens to have been wrapped inside a class, and in general use of globals should be treated with caution.  It lets you avoid passing the configuration around via method calls, but that results in the usage scope not being clearly defined, and other disadvantages associated with globals.  I won't give a -1 since it probably doesn't cause any real harm here, but I'd recommend thinking twice before using this design pattern in the future.  http://wiki.c2.com/?GlobalVariablesConsideredHarmful and http://wiki.c2.com/?GlobalVariablesAreBad are interesting reading ...